Library/Homebrew/test/test_options.rb

```diff
@@ -56,7 +56,7 @@ def test_equality
 
 class OptionsTests < Homebrew::TestCase
   def setup
-    @options = Options.new
+    @options = Options.create
```

Comment:
> Why cannot the constructor make the distinction between what is being passed to it and then act accordingly? (Sorry, I haven't found the time to carefully look at the code, but a cursory glance didn't reveal anything that would make this behavior impossible.)

We certainly can. The only question is will it hurts the readability and usage. As it will look like something like this:
```ruby
def initialize(args)
  case args
  when Array then .... # input is `["--foo=A", "--bar"]`
  when Hash then ....  # input is `{"--foo" => Option("--foo=A"), "--bar" => Option("--bar") }`
  end
end
```
I don't think we should expose this to the outside. As it's worthy to mention that `Options.create` has been there long before this PR.